### PR TITLE
test(Migration to RTL): Refactor ActivityTable.test.tsx from Enzyme to RTL

### PR DESCRIPTION
**Pull request by rtexelm**
### SUMMARY

Replace enzyme with RTL functions. Refactor the tests accordingly.

### TESTING INSTRUCTIONS

`npm test ActivityTable.test.tsx`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/features/home/ActivityTable.test.tsx [hunk 1]
```diff
@@ -85,55 +78,61 @@ describe('ActivityTable', () => {
...
-    act(() => {
-      const handler = wrapper.find('[role="tab"] a').at(1).prop('onClick');
-      if (handler) {
-        handler({} as any);
-      }
-    });
+    renderActivityTable(activityProps);
+    const editedButton = screen.getByText(/edited/i);
+    expect(editedButton).toBeInTheDocument();
+    userEvent.click(editedButton);
```
Comment (justinpark) [inline superset-frontend/src/features/home/ActivityTable.test.tsx:121]: Could you kindly make use of `import { fireEvent } from spec/helpers/testing-library` as opposed to userEvent for click events? We typically reserve userEvent for certain custom events, such as a file upload event.

```suggestion
    fireEvent.click(editedButton);
```

File: superset-frontend/src/features/home/ActivityTable.test.tsx [hunk 2]
```diff
@@ -85,55 +78,61 @@ describe('ActivityTable', () => {
...
     const dashboardCall = fetchMock.calls(/dashboard\/\?q/);
     const chartCall = fetchMock.calls(/chart\/\?q/);
-    // waitforcomponenttopaint does not work here in this instance...
+    // waitFor() does not work here in this instance...
```
Comment (justinpark) [inline superset-frontend/src/features/home/ActivityTable.test.tsx:124]: could you try with
```
await waitFor(() => expect(fetchMock.calls(/chart\/\?q/).toHaveLength(1));
```

Comment (rtexelm) [reply superset-frontend/src/features/home/ActivityTable.test.tsx:124]: that was actually my original attempt but the test will not pass, seems to be the same problem the previous author ran into. Not sure what the diagnosis is but it is passing with setTimeout

Comment (rtexelm) [reply superset-frontend/src/features/home/ActivityTable.test.tsx:124]: attempted adding a 5s timeout to the `waitFor()` with no change in result but the test passes in <20ms when using `setTimeout`. the only feedback I'm getting is a deprecation warning from the `moment` library concerning use of `non RFC2822/ISO date formats` probably from the `changed_on_utc` value in the `fetchMock`s

Comment (justinpark) [reply superset-frontend/src/features/home/ActivityTable.test.tsx]: @rtexelm it appears that the API fetch is only triggered when the 
`activeChild` prop is updated to `TableTab.Edited`.
Hence, it's being called before the case, which seems to be incorrect.

https://github.com/apache/superset/blob/47ae9d4cc3ca94332cb14359fb8a306f91c2da60/superset-frontend/src/features/home/ActivityTable.tsx#L142-L146

Kindly consider using `rerender` with `activeChild={TableTab.Edited}` (which is anticipated to be propagated after `setActiveChild()` has been called). This method should be a right direction whether the fetch has been called correctly.

Comment (rtexelm) [reply superset-frontend/src/features/home/ActivityTable.test.tsx]: Fantastic! I didn't know the usefulness of `rerender`. That looks like it's worked. The fetchMock calls don't register but the mockData is rendered without specifying `TableTab.Edited` as a replete tab

**Comment by rtexelm** [comment]
Added a small change in `ChartTable.test` that was a leftover from merging `ActivityTable.test`. It just moved `fireEvent` to `userEvent`

**Comment by rtexelm** [comment]
@justinpark is this looking better?

**Review by mistercrunch — APPROVED** [review]
LGTM

**Comment by mistercrunch** [comment]
looks like you'll have to rebase, we address the issue with the missing required checks.

**Review by justinpark — COMMENTED** [review]
LGTM!

**Pull request merged**